webpack/InsightsCloudSync/Components/InsightsSettings/InsightsSettingsReducer.js

```diff
@@ -0,0 +1,40 @@
+import Immutable from 'seamless-immutable';
+import {
+  INSIGHTS_SYNC_SETTING_SET,
+  INSIGHTS_SYNC_SETTING_SET_ERROR,
+  INSIGHTS_SYNC_SETTINGS_GET_SUCCESS,
+  INSIGHTS_SYNC_SETTINGS_GET_FAILURE,
+} from './InsightsSettingsConstants';
+
+const initialState = Immutable({
+  insightsSyncEnabled: false,
+});
+
+export default (state = initialState, action) => {
+  const { payload: { settings: insightsSyncEnabled } = {} } = action;
+
+  switch (action.type) {
+    case INSIGHTS_SYNC_SETTINGS_GET_SUCCESS:
+      return state.merge({
+        ...state,
+        insightsSyncEnabled,
+      });
+    case INSIGHTS_SYNC_SETTINGS_GET_FAILURE:
+      return state.merge({
+        ...state,
+        insightsSyncEnabled,
+      });
+    case INSIGHTS_SYNC_SETTING_SET:
+      return state.merge({
+        ...state,
+        insightsSyncEnabled,
+      });
+    case INSIGHTS_SYNC_SETTING_SET_ERROR:
+      return state.merge({
+        ...state,
+        insightsSyncEnabled,
+      });
```

Comment:
All the bodies are the same, how about
```javascript
  switch (action.type) {
    case INSIGHTS_SYNC_SETTINGS_GET_SUCCESS, INSIGHTS_SYNC_SETTINGS_GET_FAILURE, INSIGHTS_SYNC_SETTING_SET, INSIGHTS_SYNC_SETTING_SET_ERROR:
      return state.merge({
        ...state,
        insightsSyncEnabled,
      });
    default:
      return state;
  }
```

Also should the bodies be the same? When an error happens when talking to the server, shouldn't the error be taken out of the payload and put into the state?

Reply:
agree, i think we can do something like:

```js
    case INSIGHTS_SYNC_SETTINGS_GET_SUCCESS:
    case INSIGHTS_SYNC_SETTING_SET:
      return state.merge({
        ...state,
        insightsSyncEnabled,
      });
    case INSIGHTS_SYNC_SETTINGS_GET_FAILURE:
    case INSIGHTS_SYNC_SETTING_SET_ERROR:
      return state.merge({
        ...state,
        error,
      });
```

Reply:
Didn't know case in js falls through, TIL :)